xds/src/main/java/io/grpc/xds/XdsComms.java

```diff
@@ -280,6 +330,7 @@ public void onCompleted() {
           .streamAggregatedResources(xdsResponseReader);
       this.localityStore = localityStore;
 
+      cancelRetryTimer();
```

Comment:
A bit weird for an inner class to mutate outer class' state in the constructor.
Might be better to move it to `refreshAdsStream()`, and let the retry timer call `refreshAdsStream()` too.

Reply:
This was exactly my first draft version. Because besides `refreshAdsStream()`, the `XdsComms` constructor also calls `new AdsStream()`, I wanted it to be always safe to call `new AdsStream()` so I put `cancelRetryTimer()` in `AdsStream()`. Reverted.